tests/test_facebook_discovery.py

```diff
@@ -84,40 +84,26 @@ def test_run(self):
                                 msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
                                 "\nstream_properties | {}".format(stream_properties))
 
-                # BUG_1 | https://stitchdata.atlassian.net/browse/SRCE-4855
-                failing_with_no_replication_keys = {
-                    'ads_insights_country', 'adsets', 'adcreative', 'ads', 'ads_insights_region',
-                    'campaigns', 'ads_insights_age_and_gender', 'ads_insights_platform_and_device',
-                    'ads_insights_dma', 'ads_insights', 'leads', 'ads_insights_hourly_advertiser'
-                }
-                if stream not in failing_with_no_replication_keys:  # BUG_1
-                    # verify replication key(s) match expectations
-                    self.assertSetEqual(
-                        expected_replication_keys, actual_replication_keys
-                    )
+                # verify replication key(s) match expectations
+                self.assertSetEqual(
+                    expected_replication_keys, actual_replication_keys
+                )
 
                 # verify primary key(s) match expectations
                 self.assertSetEqual(
                     expected_primary_keys, actual_primary_keys,
                 )
 
-                # BUG_2 | https://stitchdata.atlassian.net/browse/SRCE-4856
-                failing_with_no_replication_method = {
-                    'ads_insights_country', 'adsets', 'adcreative', 'ads', 'ads_insights_region',
-                    'campaigns', 'ads_insights_age_and_gender', 'ads_insights_platform_and_device',
-                    'ads_insights_dma', 'ads_insights', 'leads', 'ads_insights_hourly_advertiser'
-                }
-                if stream not in failing_with_no_replication_method:  # BUG_2
-                    # verify the replication method matches our expectations
-                    self.assertEqual(
-                        expected_replication_method, actual_replication_method
-                    )
-
-                    # verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
-                    if actual_replication_keys:
-                        self.assertEqual(self.INCREMENTAL, actual_replication_method)
-                    else:
-                        self.assertEqual(self.FULL_TABLE, actual_replication_method)
+                # verify the replication method matches our expectations
+                self.assertEqual(
+                    expected_replication_method, actual_replication_method
+                )
+
```

Comment:
@savan-chovatiya What is the difference between the check you have done at line 98 and line 103?
In both the checks you are checking the value of `actual_replication_method`

Reply:
We have checked that if all stream replication methods are the same as the expected one at line 98. While we checked that if we have a replication key for any stream then it must be `INCREMENTAL` stream at line 103.

Reply:
But isn't that already being checked at line 98?
At line 98,
Stream A -> Expected replication method = "Incremental", Actual replication method = "Incremental" ===> True
Stream B -> Expected replication method = "FULL_TABLE", Actual replication method = "FULL_TABLE" ===> True

At line 103, you are checking the same thing, but more specifically
Stream A -> If replication key -> Actual replication method == "Incremental"  ==> True
Stream B -> Not verifying anything here.

In a nutshell, I believe check at line 103 is not required. Let me know if you think otherwise

Reply:
These checks are part of standard tests defined by Stitch and also these are already there in a test, we only moved it out from the if condition as we have fixed this issue for some of the streams.

Reply:
Savan is correct. This assertion will be redundant only if expectations are correct. It's real value is to catch invalid expectations set by a tester.